haystack/reader/base.py

```diff
@@ -47,8 +52,10 @@ def _calc_no_answer(no_ans_gaps: Sequence[float], best_score_answer: float):
         return no_ans_prediction, max_no_ans_gap
 
     def run(self, query: str, documents: List[Document], top_k_reader: Optional[int] = None, **kwargs): # type: ignore
+        self.query_count += 1
```

Comment:
I like the timiing in the BaseReader, but should we not add the query incrementer in the Evaluator. What if we have the reader run twice in a pipeline?

Reply:
n_queries now has to be supplied separately rather than being counted by the evaluator